### ISSUE TITLE
LDAP Location Based OU's not working as expected

### ISSUE THREAD

**Issue by michaelc64** [snipe/snipe-it#3963]
#### Expected Behavior (or desired behavior if a feature request)

Create multiple locations with different OU's
Be able to sync each one and have in sync the users in that ou.

-----

#### Actual Behavior

Issue 1: only the first 17 users in a location are synchronized
Issue 2: only the last created location is sync'd - even when specifying location on command line.

You can look at the screenshots attached.  One shows the ou settings in the locations table, next shows attempts to sync.  Even though a different location is specified it is only syncing the location with the highest id and also only the first 17 users from there.  If I add another location it then sync's the first 17 from that new one, and does not do anything with the rest.

-----

#### Please confirm you have done the following before posting your bug report:

- [Yes ] I have enabled debug mode (although nothing is reported to the log)
- [ Yes] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running - 4 beta 5
- What OS and web server you're running Snipe-IT on Windows 2016 / IIS
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) - manual
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database - no

![snipe1](https://user-images.githubusercontent.com/1715649/30463502-b509bf3c-99fe-11e7-9b1f-75d1abf0b0eb.PNG)
![snipe2](https://user-images.githubusercontent.com/1715649/30463506-b7180810-99fe-11e7-821a-20c6193995c5.PNG)

**Comment by snipe** [comment]
Can you try with the latest release and let me know if your issue is fixed?

**Comment by michaelc64** [comment]
Sort of!  I'm not sure if it is 'by design' or not but whichever location I specify in the ldap-sync command, it is still trying to sync all locations, starting with the most recently entered location.

I did however find out why it was limiting to 17 users - was due to my base-dn setting - in LdapSync.php it looks like it is looping based on number of users it finds in the global base-dn setting, rather than the number of users in the location ou.  My base dn was set to an ou with a small number of users (only those that needed to login).  If I set it back to the root of the domain it is ok.

thanks
